pwiz_tools/Skyline/Controls/Graphs/DetectionsPlotPane.cs

```diff
@@ -0,0 +1,429 @@
+        {
+
+            private DetectionsPlotPane _parent;
+            // ReSharper disable once RedundantDefaultMemberInitializer
+            private bool _isVisible = false;
+            private NodeTip _tip;
+            private TableDesc _table;
+            private readonly RenderTools _rt = new RenderTools();
+
+            //Location is in user coordinates. 
+            public int ReplicateIndex { get; private set; }
+            public bool IsVisible => _isVisible;
+
+            public ToolTipImplementation(DetectionsPlotPane parent)
+            {
+                _parent = parent;
+            }
+
+            public ITipProvider TipProvider { get { return this; } }
+
+            bool ITipProvider.HasTip => true;
+
+            Size ITipProvider.RenderTip(Graphics g, Size sizeMax, bool draw)
+            {
+                var size = _table.CalcDimensions(g);
+                if (draw)
+                    _table.Draw(g);
+                return new Size((int)size.Width + 2, (int)size.Height + 2);
+            }
+
+            public void AddLine(string description, string data)
+            {
+                if(_table == null) _table = new TableDesc();
+                _table.AddDetailRow(description,data, _rt);
+            }
+
+            public void ClearData()
+            {
+                _table?.Clear();
+            }
+
+            public void Draw(int dataIndex, Point cursorPos)
+            {
+                if (_isVisible && ReplicateIndex != dataIndex) Hide();
+                if (_isVisible && ReplicateIndex == dataIndex) return;
+                if (_table == null || _table.Count == 0) return;
+
+                ReplicateIndex = dataIndex;
+                var basePoint = new UserPoint(dataIndex + 1, 
+                    _parent.GetDataSeries()[ReplicateIndex] / _parent.YScale, _parent);
+
+                using (var g = _parent.GraphSummary.GraphControl.CreateGraphics())
+                {
+                    var size = _table.CalcDimensions(g);
+                    var offset = new Size(0, -(int)(size.Height + size.Height/_table.Count));
+                    if (_tip == null)
+                        _tip = new NodeTip(_parent);
+                    _tip.SetTipProvider(TipProvider, new Rectangle(basePoint.Screen(offset), new Size()), cursorPos);
+                }
+                _isVisible = true;
+            }
+
+            public void Hide()
+            {
+                if (_isVisible)
+                {
+                    _tip?.HideTip();
+                    _isVisible = false;
+                }
+            }
+
+            public void Dispose()
+            {
+                _rt.Dispose();
+            }
+
+            //used for testing only
+            public List<string> TipLines
+            {
+                get
+                {
+                    return _table.Select((rowDesc) =>
+                    {
+                        return Enumerable.Aggregate(

```

Comment:
It's probably simpler to use "string.Join":
                    return _table.Select(rowDesc=>
                        string.Join(TextUtil.SEPARATOR_TSV_STR, rowDesc.Select(cell=>cell.Text))
                    ).ToList();